src/Input/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+
+const getHTMLTagForType = (type) => {
+  if (type === 'select' || type === 'textarea') return type;
+  return 'input';
+};
+
+const getClassNameForType = (inputType) => {
+  if (inputType === 'file') return 'form-control-file';
+  if (inputType === 'checkbox') return 'form-check-input';
+  if (inputType === 'radio') return 'form-check-input';
+  return 'form-control';
+};
+
+const renderOptions = options => options.map(({
+  value, label, group, ...attributes
+}) => {
+  if (group) {
+    return (
+      <optgroup key={`optgroup-${label}`} label={label} {...attributes}>
+        {renderOptions(group)}
+      </optgroup>
+    );
+  }
+  return <option key={value} value={value} {...attributes}>{label}</option>;
+});
+
+
+class Input extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.innerRef = props.innerRef; // eslint-disable-line react/prop-types
```

Comment:
Good catch, thank you.

As for the eslint-disable, Since we're using forwardRef, innerRef is not a prop you can actually specify. I didn't want documentation tools to pick it up as a prop you could pass in.

Reply:
Yeah, I actually realized that after I submitted the PR review :) You can disregard my prop-types comments.